src/Arrays/KeyToVal.jl

```diff
@@ -0,0 +1,21 @@
+struct KeyToVal{T} <: Map
+  key_to_val::T
+end
+
+
+function return_cache(m::KeyToVal,keys)
```

Comment:
why the `return_cache` and `evaluate!` methods do not seem to have a last argument of the same type? 
In particular, for the `return_cache` method, it seems that we expect an array `keys`, while in the `evaluate!` method we are getting a single entry array of the `key`s array.